### PR TITLE
Email authentication

### PR DESCRIPTION
**Pull request by carolynzhang18**
## Notion Ticket Link

[Email authentication](https://www.notion.so/uwblueprintexecs/Email-Authentication-2e106a8083b44e87bf5846e2c2d06002)

## Implementation Description
* Firebase configured (OAuth Client ID)
* Nodemailer configured (.env)
* Graphql queries for checking user verification

## Steps to Test
1. Add mailer config stuff to your root .env file (check bottom of ticket on Notion 🙈)
2. Re-run docker compose up. :)
3. Go to "Sign Up" and make an account. You should get an email containing a link to verify your account. (don't click it yet!)
4. Try to log in with your credentials. This should not work because you didn't verify your email yet!
5. Now click the link to verify your email!
6. Go try to log in again. This SHOULD work because you did verify your email!

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### Mac
```bash
docker exec -it CWC-backend /bin/bash -c "yarn fix"
docker exec -it CWC-frontend /bin/bash -c "yarn fix"
```

### Windows
```bash
docker exec -it CWC-backend bash -c "yarn fix"
docker exec -it CWC-frontend bash -c "yarn fix"
```

### CODE REVIEW

**Review by achandrabalan — APPROVED** [review]
I see now that you split them by Queries and Mutations pages, nice work!

Would you be opposed to folder = Mutations and then we have specific mutation pages for different groups of features? More of an architecture question. 

Overall super impressive !!

**Comment by carolynzhang18** [comment]
@achandrabalan Yeah that's something we should discuss for sure! My original reasoning for moving all the graphql stuff into a separate folder was for better reusability - for example, checking whether a user is verified is something we do in both Login and Signup. Stack overflow (and also ChatGPT 😋) suggested splitting them up by type of operation, i.e. mutations and queries.
You're right that it's possible those files get a bit chunky and messy over time though... We could potentially also split them up according to which part of the backend we're hitting, e.g. one for auth, one for users, etc.

File: frontend/src/graphql/Mutations.ts
[the remark this answers is not in the document]
Comment (owen-sellner) [reply frontend/src/graphql/Mutations.ts:1]: That is definitely something we should be keep in mind going forward. I think it makes sense to have the queries separate from the components but in the future we should create subdirectories so bloating does not become an issue. Currently I am thinking of a structure like the graphql folder in the backend where it is divided by categories (ex. auth, user).

**Review by owen-sellner — APPROVED** [review]
LGTM! Everything is working correctly and the Linter found no issues with the files. It is ready to merge!!!!

Also, I am talking with the VPE about getting an official email set up but that does not block this PR.

**Pull request merged**